contrib/release/prep-changelog.sh

```diff
@@ -54,6 +54,7 @@ main() {
     local ersion="$(echo $2 | sed 's/^v//')"
     local version="v$ersion"
     local old_branch="$(echo $3 | sed 's/^v//')"
+    local GITHUB_TOKEN=${GITHUB_TOKEN:-"$(gh auth token)"}
```

Comment:
What happens if `gh auth login` hasn't been executed prior?

Reply:
In that case, `gh auth login` will print `no oauth token` and exit nonzero. Since this is executed within a subshell, the exit code does not result in the outer script failing. It'll continue, just like it'll do today with no github token defined, then fail some time later in the script with an authentication error.

Perhaps these scripts could all do with a `if ! gh auth status >/dev/null; then common:exit ... ; fi` snippet to ensure this will pass..?

Reply:
makes sense to me :+1: